### PR TITLE
Add Pager component

### LINKED ISSUE #1457: UI Components: Create New Pager Component

### PR DESCRIPTION
**Pull request by samshara**
fixes https://github.com/the-deep/client/issues/1457

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -58,3 +58,5 @@ export { default as Tag } from './components/Tag';
 export type { Props as TagProps } from './components/Tag';
 export type { Props as TextAreaProps } from './components/TextArea';
 export { default as TextArea } from './components/TextArea';
+export type { Props as PagerProps } from './components/Pager';
+export { default as Pager } from './components/Pager';
```
Comment (tnagorra) [inline src/index.tsx]: Exports should be sorted in alphabetical order

File: src/components/Pager/index.tsx [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    itemsPerPageControlHidden?: false;
+    onItemsPerPageChange: (pageCapacity: number) => void;
+})
+
+const defaultOptions: PagerOption[] = [
+    { label: '10 / page', key: 10 },
+    { label: '25 / page', key: 25 },
+    { label: '50 / page', key: 50 },
+    { label: '75 / page', key: 75 },
+    { label: '100 / page', key: 100 },
```
Comment (tnagorra) [inline src/components/Pager/index.tsx]: I think we should remove these 2 options

File: src/components/Pager/index.tsx [hunk 3]
```diff
@@ -0,0 +1,314 @@
+import React from 'react';
```
Comment (tnagorra) [inline src/components/Pager/index.tsx]: Please refer to this updated file
https://github.com/toggle-corp/toggle-ui/blob/dfb69ebe734e7a90cbd2424c2abc9c2c85848561/src/components/Pager/index.tsx

**Review by tnagorra — CHANGES_REQUESTED** [review]
Need to make Pager up-to-date with toggle-ui

File: src/stories/Pager.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import Pager, { PagerProps } from '#components/Pager';
+
+export default {
+    title: 'Input/Pager',
+    component: Pager,
+    argTypes: {},
+};
+
+const Template: Story<PagerProps> = (args) => {
+    const [{ activePage, maxItemsPerPage }, handleArgsChange] = useArgs();
```
Comment (tnagorra) [inline src/stories/Pager.stories.tsx:16]: Why read maxItemsPerPage?

**Comment by tnagorra** [comment]
Needs more work on the styling

**Review by frozenhelium — APPROVED** [review]

File: src/components/Pager/index.tsx [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+                        className={styles.pageSpan}
+                    >
+                        {page.label}
+                    </div>
+                );
+            })}
+            {showPrevNext && (
+                <Button
+                    name={undefined}
+                    onClick={() => onActivePageChange(activePage + 1)}
```
Comment (tnagorra) [inline src/components/Pager/index.tsx]: Here, you can use `name={activePage + 1}`

File: src/components/Pager/index.tsx [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+            </div>
+            <div className={styles.total}>
+                {itemsCount}
+            </div>
+        </div>
+    );
+
+    const itemsPerPageOptions = options ?? defaultOptions;
+
+    const minOption = Math.min(0, ...itemsPerPageOptions.map((opt) => opt.key));
```
Comment (tnagorra) [inline src/components/Pager/index.tsx]: maybe we should memoize this

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**